test/kafo/hiera_configurer_test.rb

```diff
@@ -15,7 +15,7 @@ module Kafo
     end
 
     describe "#generate_data" do
-      let(:puppet_module) { PuppetModule.new('testing', TestParser.new(BASIC_MANIFEST)).tap { |m| m.enable }.parse }
+      let(:puppet_module) { @@puppet_module ||= PuppetModule.new('testing', TestParser.new(BASIC_MANIFEST)).tap { |m| m.enable }.parse }
```

Comment:
Should the caching live inside TestParser? Some of the test factories already have class-level caches.